Building with -O0 causes programs/zstd to fail to link

For example, configuring the meson build using

```
$ meson setup --buildtype=debug builddir/
$ ninja -C builddir/
[50/50] Linking target programs/zstd
FAILED: programs/zstd 
cc  -o programs/zstd programs/zstd.p/.._.._.._programs_zstdcli.c.o programs/zstd.p/.._.._.._programs_util.c.o programs/zstd.p/.._.._.._programs_timefn.c.o programs/zstd.p/.._.._.._programs_fileio.c.o programs/zstd.p/.._.._.._programs_benchfn.c.o programs/zstd.p/.._.._.._programs_benchzstd.c.o programs/zstd.p/.._.._.._programs_datagen.c.o programs/zstd.p/.._.._.._programs_dibio.c.o programs/zstd.p/.._.._.._programs_zstdcli_trace.c.o programs/zstd.p/.._.._.._lib_common_xxhash.c.o -Wl,--as-needed -Wl,--no-undefined '-Wl,-rpath,$ORIGIN/../lib' -Wl,-rpath-link,/tmp/zstd-1.5.1/build/meson/builddir/lib -Wl,--start-group lib/libzstd.so.1.5.1 /usr/lib/libz.so /usr/lib/liblzma.so /usr/lib/liblz4.so -Wl,--end-group
/usr/bin/ld: programs/zstd.p/.._.._.._programs_dibio.c.o: in function `ERR_getErrorName':
/tmp/zstd-1.5.1/build/meson/builddir/../../../programs/../lib/common/error_private.h:75: undefined reference to `ERR_getErrorString'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

I can reproduce the failure by setting the buildtype=plain, and make the error go away by additionally adding `CFLAGS=-O1`.

Although presumably zstd may rather suffer for building without any optimizations, undefined symbols seems odd.

--- a/programs/benchzstd.c
+++ b/programs/benchzstd.c
@@ -31,9 +31,14 @@
 #include "timefn.h"      /* UTIL_time_t */
 #include "benchfn.h"
 #include "../lib/common/mem.h"
+#ifndef ZSTD_STATIC_LINKING_ONLY
 #define ZSTD_STATIC_LINKING_ONLY
+#endif
 #include "../lib/zstd.h"
 #include "datagen.h"     /* RDG_genBuffer */
+#ifndef XXH_INLINE_ALL
+#define XXH_INLINE_ALL
+#endif
 #include "../lib/common/xxhash.h"
 #include "benchzstd.h"
 #include "../lib/zstd_errors.h"

--- a/programs/dibio.c
+++ b/programs/dibio.c
@@ -31,7 +31,6 @@
 
 #include "timefn.h"         /* UTIL_time_t, UTIL_clockSpanMicro, UTIL_getTime */
 #include "../lib/common/mem.h"  /* read */
-#include "../lib/common/error_private.h"
 #include "dibio.h"
 
 

